src/Compiler/pars.fsy

```diff
@@ -4430,6 +4430,16 @@ declExpr:
      { let trivia: SynExprYieldOrReturnTrivia = { YieldOrReturnKeyword = rhs parseState 1 }
        SynExpr.YieldOrReturn(($1, not $1), $2, (unionRanges (rhs parseState 1) $2.Range), trivia) }
 
+  | YIELD declExpr COLON typ
```

Comment:
Please also try adding recovery rules that would apply while typing the expression, e.g.:

```fsharp
let _ =
    async {
        return new MyType() :
    }
```

Reply:
Done. Thanks for the review